### PR TITLE
Add webhook sender for Alertmanager notifications

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR adds a struct that implements the [WebhookSender interface](https://github.com/grafana/alerting/blob/bb4f4f429d364ff8e5cffd4f81005eff2132633a/receivers/webhook.go#L16-L18) describes in our Alerting package, it will be used to send HTTP requests from the Alertmanager.

### CODE REVIEW

File: pkg/alertmanager/sender.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/cortexproject/cortex/blob/master/pkg/alertmanager/distributor.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Cortex Authors.
+
+// Mostly taken from http://github.com/grafana/grafana/pkg/services/notification/webhook.go
```
Comment (stevesg) [inline pkg/alertmanager/sender.go:6]:
```suggestion
// Mostly taken from http://github.com/grafana/grafana/main/pkg/services/notifications/webhook.go
```

**Review by stevesg — APPROVED** [review]
LGTM - Approving because it's 99% the same as the Grafana code.

File: pkg/alertmanager/sender.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```
Comment (gotjosh) [inline pkg/alertmanager/sender.go:1]: Out of curiosity, couldn't we technically copy this directly into grafana/alerting? With that we could have both Grafana and Mimir use the same code as opposed to a copy that could potentially diverge?

Comment (santihernandezc) [reply pkg/alertmanager/sender.go:1]: This code is not in the alerting package. The interface is defined there, bt the implementations is up to each importing package.

**Review by stevesg — APPROVED** [review]

**Review by gotjosh — APPROVED** [review]

**Pull request merged**